cmd/gateway-azure.go

```diff
@@ -639,14 +656,47 @@ func (a *azureObjects) CompleteMultipartUpload(bucket, object, uploadID string,
 	if meta == nil {
 		return objInfo, traceError(InvalidUploadID{uploadID})
 	}
+
+	resp, err := a.client.GetBlockList(bucket, object, storage.BlockListTypeUncommitted)
+	if err != nil {
+		return objInfo, azureToObjectError(traceError(err), bucket, object)
+	}
+
+	getBlocks := func(partNumber int, etag string) (blocks []storage.Block, err error) {
+		var idBytes []byte
+		for _, part := range resp.UncommittedBlocks {
+			if idBytes, err = base64.StdEncoding.DecodeString(part.Name); err != nil {
+				return nil, err
+			}
+
+			id := string(idBytes)
+			if strings.HasPrefix(id, strconv.Itoa(partNumber)+".") && strings.HasSuffix(id, "."+etag) {
+				blocks = append(blocks, storage.Block{
+					ID:     part.Name,
+					Status: storage.BlockStatusUncommitted,
+				})
+			}
+		}
+
+		if len(blocks) == 0 {
+			return nil, InvalidPart{}
+		}
+
+		return blocks, nil
+	}
+
+	var allBlocks []storage.Block
 	var blocks []storage.Block
```

Comment:
blocks can be inside the scope of the `for loop` below

Reply:
Done